llvm/test/MC/Disassembler/AMDGPU/gfx8_mimg_features.txt

```diff
@@ -235,5 +235,5 @@
 # VI: image_gather4 v[252:255], v1, s[8:15], s[12:15] dmask:0x3 ; encoding: [0x00,0x03,0x00,0xf1,0x01,0xfc,0x62,0x00]
 0x00,0x03,0x00,0xf1,0x01,0xfc,0x62,0x00
 
-# VI: image_gather4 v[252:255], v1, s[8:15], s[12:15] dmask:0x1 unorm glc slc tfe lwe da ; encoding: [0x00,0x71,0x03,0xf3,0x01,0xfc,0x62,0x00]
+# VI: image_gather4 v[252:253], v1, s[8:15], s[12:15] dmask:0x1 unorm glc slc tfe lwe da ; encoding: [0x00,0x71,0x03,0xf3,0x01,0xfc,0x62,0x00]
```

Comment:
Why does this change?

Reply:
Previous default was V4 variant so it would start with VReg_128: v[252:255]
Now default is V2 so it starts with VReg_64: v[252:253]
And I think both fail when trying to convert to V5 and VReg_160: v[252:**256**] which would be invalid.
I'm not sure what the original test was checking since this is one of "incorrect" encodings

Reply:
Based on comment at line 231 it was probably testing that with tfe it will go out of range

Reply:
Based on comment at line 231 it was probably testing that with tfe it will go out of range.

Reply:
So really this should fail to disassemble, in the same way as any other instruction that uses a vgpr tuple that goes higher than v255. Right?

Reply:
This currently matches sp3. It does fail on encoding as does sp3.

Looking into it further it seems that we do have inconsistent behavior.

We do not decode VOP when a register is out of bounds (gfx1010 example): 
```
# v_add_f64 v[255:256], v[1:2], v[2:3]
0xff,0x00,0x64,0xd5,0x01,0x05,0x02,0x00
```

But we do MIMG (gfx1010 example):
```
# image_load v65, v[253:256], s[0:7] dmask:0x8 dim:SQ_RSRC_IMG_2D_MSAA_ARRAY unorm
0x38,0x18,0x00,0xf0,0xfd,0x41,0x00,0x00 
```
except llvm produces just v253. Sp3 always decodes to instructions above.

Reply:
For invalid encodings, I would prefer that we consistently either fail to decode, or print something like `v[252:256] /* VGPR tuple out of range */`. But that would be for another patch.

I have no objection to the current patch if it fixes disassembly of valid encodings.